pkgs/development/libraries/blst/default.nix

```diff
@@ -0,0 +1,30 @@
+{ stdenv, lib, fetchFromGitHub, autoreconfHook }:
+
+stdenv.mkDerivation ( finalAttrs: {
+  name = "blst-0.3.10";
+
+  src = fetchFromGitHub {
+    owner = "supranational";
+    repo = "blst";
+    rev = "03b5124029979755c752eec45f3c29674b558446";
+    hash = "01m28xxmm1x7riwyax7v9rycwl5isi06h2b2hl4gxnnylkayisn5";
```

Comment:
I'm getting a `error: cannot find flake 'flake:blst' in the flake registries`. Do you know why that might be?

Reply:
Here is the set of commands I used to test that this PR still works:
```
yvan@X230 ~ % gh repo clone iquerejeta/nixpkgs
yvan@X230 ~ % export NIX_PATH=nixpkgs=$(pwd)/nixpkgs
yvan@X230 ~ % nix repl                   
Welcome to Nix 2.13.3. Type :? for help.

nix-repl> pkgs = import <nixpkgs> {}
nix-repl> pkgs.libblst               
error: hash '01m28xxmm1x7riwyax7v9rycwl5isi06h2b2hl4gxnnylkayisn5' does not include a type, nor is the type otherwise known from context
«derivation 
```
You should have written ` hash = "sha256:01m28xxmm1x7riwyax7v9rycwl5isi06h2b2hl4gxnnylkayisn5";` (with `sha256` prefix) 😉

Reply:
`nix build blst` isn't the right way of triggering the evaluation of the derivation … but the technic that mentioned @drupol to obtain the hash: TOFU ([Trust on first use](https://en.wikipedia.org/wiki/Trust_on_first_use)) as _“Trust the hash of a source for the first time when you fetch it, and fix it in the nix expression”_ is the right way of obtaining an SRI hash without bothering about understanding its format! Just use an empty string and iterate from the error! 😄

Reply:
Really? I made the change I suggested and then everything works as expected:

```
yvan@X230 ~ % nix repl
Welcome to Nix 2.13.3. Type :? for help.

nix-repl> pkgs = import <nixpkgs> {} 

nix-repl> pkgs.libblst      
«derivation /nix/store/gb2i3jf25ir5d4v0z47s1jlpz9al32yw-blst-0.3.10.drv»

yvan@X230 ~ % nix-build /nix/store/gb2i3jf25ir5d4v0z47s1jlpz9al32yw-blst-0.3.10.drv
this derivation will be built:
  /nix/store/gb2i3jf25ir5d4v0z47s1jlpz9al32yw-blst-0.3.10.drv
these 2 paths will be fetched (0.44 MiB download, 3.33 MiB unpacked):
  /nix/store/c3f4jdwzn8fm9lp72m91ffw524bakp6v-stdenv-linux
  /nix/store/ghxnazmksdqhwynzbkqikaa6bikla655-source
copying path '/nix/store/ghxnazmksdqhwynzbkqikaa6bikla655-source' from 'https://cache.zw3rk.com'...
copying path '/nix/store/c3f4jdwzn8fm9lp72m91ffw524bakp6v-stdenv-linux' from 'https://cache.nixos.org'...
building '/nix/store/gb2i3jf25ir5d4v0z47s1jlpz9al32yw-blst-0.3.10.drv' on 'ssh://builder@x86_64-linux-2.zw3rk.com'...
copying path '/nix/store/wyzaa007anaxxhmrcbffm597v14n2mxs-linux-headers-6.1' from 'https://cache.nixos.org'...
copying path '/nix/store/6g6d4la7xsizvr8qg91f56jiqx149iqq-binutils-2.40-lib' from 'https://cache.nixos.org'...
copying path '/nix/store/45a55rzx3k794626g8adslzc6557gh0j-expand-response-params' from 'https://cache.nixos.org'...
copying path '/nix/store/fhzz4yrdy17czwc9i4swhlpcp445inzb-binutils-2.40' from 'https://cache.nixos.org'...
copying path '/nix/store/pqnd39aq2sksad2zvswjcpkqdc7ig3f9-glibc-2.35-224-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/wx4nsqbssywgx9zkabwgl9sxm5l61ni3-gcc-12.2.0' from 'https://cache.nixos.org'...
copying path '/nix/store/i2pdyabq6nrrnisbkma71h42fw6ha0y6-binutils-wrapper-2.40' from 'https://cache.nixos.org'...
copying path '/nix/store/bfbp3ypd9nm3fapz634gvvs738blrl0y-gcc-wrapper-12.2.0' from 'https://cache.nixos.org'...
copying path '/nix/store/c3f4jdwzn8fm9lp72m91ffw524bakp6v-stdenv-linux' from 'https://cache.nixos.org'...
copying 0 paths...
unpacking sources
unpacking source archive /nix/store/ghxnazmksdqhwynzbkqikaa6bikla655-source
source root is source
patching sources
configuring
no configure script, doing nothing
building
+ gcc -O -fno-builtin -fPIC -Wall -Wextra -Werror -D__ADX__ -mno-avx -c ./src/server.c
+ gcc -O -fno-builtin -fPIC -Wall -Wextra -Werror -D__ADX__ -mno-avx -c ./build/assembly.S
+ ar rc libblst.a assembly.o server.o
running tests
no Makefile or custom checkPhase, doing nothing
installing
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/maryx6yj7lk392i1z1cxc9g3sqb8h75a-blst-0.3.10
checking for references to /build/ in /nix/store/maryx6yj7lk392i1z1cxc9g3sqb8h75a-blst-0.3.10...
patching script interpreter paths in /nix/store/maryx6yj7lk392i1z1cxc9g3sqb8h75a-blst-0.3.10
stripping (with command strip and flags -S) in  /nix/store/maryx6yj7lk392i1z1cxc9g3sqb8h75a-blst-0.3.10/lib
copying 1 paths...
copying path '/nix/store/maryx6yj7lk392i1z1cxc9g3sqb8h75a-blst-0.3.10' from 'ssh://builder@x86_64-linux-2.zw3rk.com'...
/nix/store/maryx6yj7lk392i1z1cxc9g3sqb8h75a-blst-0.3.10
```

Reply:
It appears both work, but yes the empty hash suggest to rather to use `-` separator:

```
% nix repl                                                             
Welcome to Nix 2.13.3. Type :? for help.

nix-repl> pkgs = import <nixpkgs> {}

nix-repl> pkgs.libblst               
warning: found empty hash, assuming 'sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA='
«derivation /nix/store/9p5xb8ygbgf2pf2mq1w1vhrcghflx8wd-blst-0.3.10.drv»

yvan@X230 ~/nixpkgs (git)-[master] % nix-build /nix/store/9p5xb8ygbgf2pf2mq1w1vhrcghflx8wd-blst-0.3.10.drv
these 2 derivations will be built:
  /nix/store/8vdc08f3b8cs5i1zkrpzflk7aw7fcrwa-source.drv
  /nix/store/9p5xb8ygbgf2pf2mq1w1vhrcghflx8wd-blst-0.3.10.drv
these 20 paths will be fetched (4.34 MiB download, 18.12 MiB unpacked):
  /nix/store/0k5dhghrssfd0fs0bkvaw698w6666v9g-mirrors-list
  /nix/store/2nmqzvgf4rnjvpjsdkycf41lf63br42a-unzip-6.0
  /nix/store/2zc9s4cbdq1cnzimhqj5mv71qa3xpcp7-libkrb5-1.20.1-dev
  /nix/store/7vwvn3d5yb5yb3vdv9iv4321ckx4paf9-nss-cacert-3.86
  /nix/store/8sqpilc7xz92rs2d3i2l54ahhbpir1jg-curl-7.88.0-bin
  /nix/store/aq06nz8mq5nxkccasnj94zdk9pfqi2vs-openssl-3.0.8
  /nix/store/dawkj271851d8kpxz97grfsryjzlhsjj-glibc-locales-2.35-224
  /nix/store/f0j5grk3yf8pnml676jrcph4a88g8c2g-curl-7.88.0-dev
  /nix/store/facpy8mrdga06s4vcfzampl1b3rdh2kx-libssh2-1.10.0
  /nix/store/hf066cp95jiq13xl46s2zrdi94qvqlvv-openssl-3.0.8-dev
  /nix/store/hixx5vvivfzwj8cil0a5sfxbqp7lvb1d-nghttp2-1.51.0
  /nix/store/hwka9k94awxf8yq99rp7x0zbwksha5a7-curl-7.88.0
  /nix/store/jij4waw18na5smksrxfwzky7nd0hf2m6-curl-7.88.0-man
  /nix/store/mm2g3f5s75618ck1q4niyl1k57vx88ha-nghttp2-1.51.0-bin
  /nix/store/mpds2vrlb6m0yxs23fi6gs4z7y7h7ljr-nghttp2-1.51.0-lib
  /nix/store/qgqsyg7gmqy3fz6shvzhcc7m7pkfw2ig-zlib-1.2.13-dev
  /nix/store/rwfwqrvs6sjflbzc8wnkj57mnv94s40s-libkrb5-1.20.1
  /nix/store/ylgmlavny0m9afvqmki0w7wciprhvk69-nghttp2-1.51.0-dev
  /nix/store/zfsf3gffd6myvlkly4hb4cdndbb7dyic-openssl-3.0.8-bin
  /nix/store/zs3v6nqczpgs8v9pnk528p6kkr8h46f2-libssh2-1.10.0-dev
copying path '/nix/store/dawkj271851d8kpxz97grfsryjzlhsjj-glibc-locales-2.35-224' from 'https://cache.nixos.org'...
copying path '/nix/store/0k5dhghrssfd0fs0bkvaw698w6666v9g-mirrors-list' from 'https://cache.nixos.org'...
copying path '/nix/store/7vwvn3d5yb5yb3vdv9iv4321ckx4paf9-nss-cacert-3.86' from 'https://cache.nixos.org'...
copying path '/nix/store/jij4waw18na5smksrxfwzky7nd0hf2m6-curl-7.88.0-man' from 'https://cache.nixos.org'...
copying path '/nix/store/rwfwqrvs6sjflbzc8wnkj57mnv94s40s-libkrb5-1.20.1' from 'https://cache.nixos.org'...
copying path '/nix/store/hixx5vvivfzwj8cil0a5sfxbqp7lvb1d-nghttp2-1.51.0' from 'https://cache.nixos.org'...
copying path '/nix/store/mm2g3f5s75618ck1q4niyl1k57vx88ha-nghttp2-1.51.0-bin' from 'https://cache.nixos.org'...
copying path '/nix/store/mpds2vrlb6m0yxs23fi6gs4z7y7h7ljr-nghttp2-1.51.0-lib' from 'https://cache.nixos.org'...
copying path '/nix/store/aq06nz8mq5nxkccasnj94zdk9pfqi2vs-openssl-3.0.8' from 'https://cache.nixos.org'...
copying path '/nix/store/2nmqzvgf4rnjvpjsdkycf41lf63br42a-unzip-6.0' from 'https://cache.nixos.org'...
copying path '/nix/store/qgqsyg7gmqy3fz6shvzhcc7m7pkfw2ig-zlib-1.2.13-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/2zc9s4cbdq1cnzimhqj5mv71qa3xpcp7-libkrb5-1.20.1-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/ylgmlavny0m9afvqmki0w7wciprhvk69-nghttp2-1.51.0-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/zfsf3gffd6myvlkly4hb4cdndbb7dyic-openssl-3.0.8-bin' from 'https://cache.nixos.org'...
copying path '/nix/store/facpy8mrdga06s4vcfzampl1b3rdh2kx-libssh2-1.10.0' from 'https://cache.nixos.org'...
copying path '/nix/store/zs3v6nqczpgs8v9pnk528p6kkr8h46f2-libssh2-1.10.0-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/hwka9k94awxf8yq99rp7x0zbwksha5a7-curl-7.88.0' from 'https://cache.nixos.org'...
copying path '/nix/store/hf066cp95jiq13xl46s2zrdi94qvqlvv-openssl-3.0.8-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/8sqpilc7xz92rs2d3i2l54ahhbpir1jg-curl-7.88.0-bin' from 'https://cache.nixos.org'...
copying path '/nix/store/f0j5grk3yf8pnml676jrcph4a88g8c2g-curl-7.88.0-dev' from 'https://cache.nixos.org'...
building '/nix/store/8vdc08f3b8cs5i1zkrpzflk7aw7fcrwa-source.drv'...

trying https://github.com/supranational/blst/archive/03b5124029979755c752eec45f3c29674b558446.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  526k    0  526k    0     0   779k      0 --:--:-- --:--:-- --:--:--  779k
unpacking source archive /build/03b5124029979755c752eec45f3c29674b558446.tar.gz
error: hash mismatch in fixed-output derivation '/nix/store/8vdc08f3b8cs5i1zkrpzflk7aw7fcrwa-source.drv':
         specified: sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=
            got:    sha256-xero1aTe2v4IhWIJaEDUsVDOfE77dOV5zKeHWntHogY=
error: 1 dependencies of derivation '/nix/store/9p5xb8ygbgf2pf2mq1w1vhrcghflx8wd-blst-0.3.10.drv' failed to build
```

Reply:
Also, hash is different 🤔 damn, I do not understand SRI …

Reply:
AFAIU `-` is the syntax of non SRI-encoded hash to nix https://github.com/NixOS/nix/blob/master/src/libutil/hash.cc#L123 and `:` means SRI-encoded so `base64`. Am I right?

Reply:
Ok I'm back.

First of all, sorry @iquerejeta , I gave you an wrong information by telling you to run `nix build blst`, it actually should have been `nix build .#libblst`.

Second, regarding the hash, I don't mind using `:` or `-`, but I do care about consistency. In `nixpkgs`, I've never seen anything else than `-`, so I would have the tendency to follow that pattern.

And lastly, thanks for implementing all the changes, everything seems to be ok now :)